netty/src/main/java/io/grpc/netty/NettyServerStream.java

```diff
@@ -206,6 +204,29 @@ public void deframeFailed(Throwable cause) {
       handler.getWriteQueue().enqueue(new CancelServerStreamCommand(this, status), true);
     }
 
+    private void handleWriteFutureFailures(ChannelFuture future) {
+      // isStreamDeallocated() check protects from spamming stream resets by scheduling multiple
+      // CancelServerStreamCommand commands.
+      if (future.isSuccess() || isStreamDeallocated()) {
+        return;
+      }
+      // Future failed, fail RPC.
+      // Normally we don't need to do anything here because the cause of a failed future
+      // while writing DATA frames would be an IO error and the stream is already closed.
+      // However, we still need handle any unexpected failures raised in Netty.
+      // TODO(sergiitk): check if something similar to
```

Comment:
TODO: discuss with @ejona86

Reply:
From IRL: doesn't matter

Reply:
For posterity: it doesn't matter much on server-side because few things will see this error message. We just report "cancelled!" to the server. The status code shows up in metrics, but not the messages. We do use them in debugging, but we really don't expect this to trigger except if there is a bug, so it isn't much of a concern.

Reply:
Thanks for the note!